### DISCUSSION TITLE
Not clear how to use custom prompt

### DISCUSSION THREAD

**Discussion by jppaolim** [longy2k/obsidian-bmo-chatbot#52] [General]
Amazing work, but I am not clear on how to use them. 
Let's say I have a directory with interesting prompt. One prompt is, for instance,  "summarize this", how can use it to work on a note ... Shall I reference active note and then say "Go" ? Shall I copy paste the current note into the chat ?  I am sorry but I can't seem to understand how you use the custom prompts ...  
Similarly when you say, reference the active note, is it pasted in the context at the end of the prompt ? Before ? how does it work ... thanks !!

**Comment by longy2k** [comment]
The 'Reference Current Note' and 'Prompt' feature is added to the system role (e.g. System: `referenceCurrentNoteContent + system_role + prompt`). 

Adding the referenceCurrentNoteContent and prompt content into the system role is not the best way of implementing these features because some models might ignore some/all of the requirements, or the models might forget it over time. I am currently learning about embeddings, function callings, and etc to resolve this problem.

When you have the 'Reference Current Note' activated and you want to summarize the note, it would be best to just write something like "Summarize my note." You should be specific with your responses so that the model knows what to look for.

When creating prompts, the prompt is appended to the system role which will influence how the model responds. 

For example, you can create a prompt that says, "Act like Santa Claus", the model will respond with a personality like Santa Claus.

Another prompt that I find useful at times is, "Respond only in bullet points", that way, when I chat with the models, I don't have to remind it to respond in bullet points each time.

I hope this help clarify some of your questions.

If you're still confused, let me know, and I can provide some more examples!

Thanks

**Comment by motion2082** [comment]
I'm a little confused about System Prompts also.

I have a folder called "BMO/Prompts" 

How are prompts stored in this folder? As .md notes? 

To make a prompt "summarise the following selected text" would you just write that in a note then CTRL+P BMO Prompt Select Generate as when I did this /prompt is still empty
